koiki/client.py

```diff
@@ -24,34 +23,49 @@ def create_delivery(self):
         endpoint_req = CreateDelivery(self.order)
         req_body = self._authentication(endpoint_req)
         url = self._url(endpoint_req)
-        self.logger.info('Koiki request to {}. body={}'.format(url, req_body))
+        self.logger.info("Koiki request to {}. body={}".format(url, req_body))
 
         response = requests.post(url, json=req_body)
         response_body = json.loads(response.text)
 
         deliveries = []
         if response.status_code == status.HTTP_200_OK:
-            for num, delivery_data in enumerate(response_body['envios']):
+            for num, delivery_data in enumerate(response_body["envios"]):
                 vendor = self.order.vendors[num]
-                req_body_shipping = req_body['envios'][num]
-                delivery_data['order_id'] = self.order.order_id
+                req_body_shipping = req_body["envios"][num]
+                delivery_data["order_id"] = self.order.order_id
                 delivery = Delivery(delivery_data, vendor, req_body_shipping)
+                delivery_pdf = None
+
                 if delivery._is_errored():
-                    self._log(delivery.data.get('respuesta'), delivery.data, level='error')
+                    self._log(delivery.data.get("respuesta"), delivery.data, level="error")
+                    status_delivery = DeliveryStatus.ERROR
                 else:
-                    self._log(delivery.data.get('respuesta'), self._masked_body(delivery_data))
+                    self._log(delivery.data.get("respuesta"), self._masked_body(delivery_data))
+                    status_delivery = DeliveryStatus.SUCCESS
+                    delivery_pdf = delivery.print_pdf()
 
                 deliveries.append(delivery)
+                PersistentDelivery(
+                    shipment=delivery_data["numPedido"],
```

Comment:
Rather than using the word "order_number" shound't be better to call it shipment_number ?  Since koiki only deals with shipments...  and since we are already using "order" to keep the id of the woocomerce order id...

Reply:
Also, we're opening this class to append every new feature into this method...
the idea is only to keep the relationship between WC's order, WC plugin's vendor and koiki's shipment id (still need to research their API to see if we can use that numPedido to frequently check the status of the shipment... )

Reply:
I see your point. I guess we mistakenly matched `numPedido` to the WC's order but what looks really odd (I think that's what you want to investigate) the fact we provide `numPedido` in the POST requests to `/altaEnvios`. How is this going to be the delivery's unique ID?

By looking at the _API Seguimiento_ docs, it seems that the way to uniquely identify deliveries is with the bar code.

Reply:
> the idea is only to keep the relationship between WC's order, WC plugin's vendor and koiki's shipment id

these are already too many responsibilities which will require changing this class almost in every PR :warning:

Reply:
> these are already too many responsibilities which will require changing this class almost in every PR warning

I don't understand what do you mean here. The only responsability is to keep the relationship for persistence and in the future other classes may use this of this model with other responsabilities(ex, check status of shipment)

Reply:
what I mean is that that's already a lot to deal with and that is why we've been changing this class so often lately. The responsibility you mention is yet another one. This was initially intended to deal with transport logic, such as building the HTTP request., so that makes them two already.

Reply:
But this class is a new class for the Django model, not the Delivery class or the Shipment class we already have.

Reply:
Sorry for the confusion but the comment

> Also, we're opening this class to append every new feature into this method, and that's a very clear code smell already.

was meant for this specific class we're commenting on: `koiki.Client`.